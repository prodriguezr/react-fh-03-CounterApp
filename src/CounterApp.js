import PropTypes from 'prop-types';
import { useState } from 'react';

const CounterApp = ({ initialValue = 10 }) => {
  const [counter, setCounter] = useState(initialValue);

  const handlerAdd = () => {
    setCounter(counter + 1);
  };
  const handlerReset = () => {
    setCounter(initialValue);
  };
  const handlerSubstract = () => {
    setCounter(counter - 1);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>

      <button id='btnAdd' onClick={handlerAdd}>
        + 1
      </button>
      <button id='btnReset' onClick={handlerReset}>
        RESET
      </button>
      <button id='btnSubstract' onClick={handlerSubstract}>
        - 1
      </button>
    </>
  );
};

CounterApp.propTypes = {
  initialValue: PropTypes.number,
};

export default CounterApp;
