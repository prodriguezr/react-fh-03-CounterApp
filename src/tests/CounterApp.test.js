import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('Tests on FirstApp', () => {
  const wrapper = shallow(<CounterApp />);

  test('should display the "<CounterApp/>" component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should display the initial value of 100', () => {
    const initialValue = 100;

    const wrapper = shallow(<CounterApp initialValue={initialValue} />);

    expect(wrapper.find('h2').text().trim()).toBe('100');
  });

  test('Must increase with the +1 button', () => {
    wrapper.find('#btnAdd').simulate('click');

    expect(wrapper.find('h2').text().trim()).toBe('11');
  });

  test('Must decrease with the -1 button', () => {
    wrapper.find('#btnSubstract').simulate('click');

    expect(wrapper.find('h2').text().trim()).toBe('10');
  });

  test('Must place the default value with the btnReset button', () => {
    const initialValue = 21;
    const wrapper = shallow(<CounterApp initialValue={initialValue} />);

    wrapper.find('#btnAdd').simulate('click');
    wrapper.find('#btnReset').simulate('click');

    expect(Number(wrapper.find('h2').text().trim())).toBe(initialValue);
  });
});
