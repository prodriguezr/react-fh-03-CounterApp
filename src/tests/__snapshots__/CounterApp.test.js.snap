// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on FirstApp should display the "<CounterApp/>" component correctly 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    id="btnAdd"
    onClick={[Function]}
  >
    + 1
  </button>
  <button
    id="btnReset"
    onClick={[Function]}
  >
    RESET
  </button>
  <button
    id="btnSubstract"
    onClick={[Function]}
  >
    - 1
  </button>
</Fragment>
`;
